src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
```

Comment:
On second thought, I think it's not easy to put this logic inside `amp-share-tracking.js`. Since we need to know the embed state of the viewer to determine whether to update the url fragment. We can only chech the embed state inside the `viewer-impl`